towerofhanoi/Program.cs

```diff
@@ -0,0 +1,228 @@
+                    foreach (Blocks z in stackPrinted)
+                    {
+                        System.Console.WriteLine(z.weightedBlock);
+                    }
+                    System.Console.WriteLine();
+                }
+            }
+        }
+        public string UserMoveFrom()
+        {
+            Console.WriteLine("What tower are you moving your block from? Please choose 'A'  'B' or 'C' ");
+            userChoice1 = Console.ReadLine().ToUpper().Trim();
+            return userChoice1;
+        }
+        //validates choice
+        public bool ValidMoveFrom(string userChoice1)
+        {
+            bool validMoveFrom;
+            if ((userChoice1 == "A") || (userChoice1 == "B") || (userChoice1 == "C"))
+            {
+                moveFrom = userChoice1;
+                validMoveFrom = true;
+            }
+            else
+            {
+                validMoveFrom = false;
+                throw new Exception();
+            }
+            return validMoveFrom;
+        }
+        //This is what user chooses to move to.
+        public string UserMoveTo()
+        {
+            Console.WriteLine("What tower are you moving your block to? Please choose 'A'  'B' or 'C' ");
+            userChoice2 = Console.ReadLine().ToUpper().Trim();
+            return userChoice2;
+        }
+        //validates choice
+        public bool ValidMoveTo(string userChoice2)
+        {
+            // bool validMoveTo;
+            if ((userChoice2 == "A") || (userChoice2 == "B") || (userChoice2 == "C"))
+            {
+                movedTo = userChoice2;
+                validMoveTo = true;
+            }
+            else
+            {
+                validMoveTo = false;
+                throw new Exception();
+            }
+            return validMoveTo;
+        }
+        public void MoveBlock()
+        {
+            //checks to see if from is empty
+            if (MakeTowers[moveFrom].towerBlocks.Count == 0)
+            {
+                throw new Exception();
+            }
+            var MovingPiece = MakeTowers[moveFrom].towerBlocks.Peek();
+            //check for moved To if empty
+            if (MakeTowers[movedTo].towerBlocks.Count == 0)
+            {
+                MakeTowers[moveFrom].towerBlocks.Pop();
+                MakeTowers[movedTo].towerBlocks.Push(MovingPiece);
+                //check if block in moved to is bigger
+            }
+            else if (MakeTowers[movedTo].towerBlocks.Peek().weightedBlock > MovingPiece.weightedBlock)
+            {
+                MakeTowers[moveFrom].towerBlocks.Pop();
+                MakeTowers[movedTo].towerBlocks.Push(MovingPiece);
+            }
+            else
+            {
+                throw new Exception();
+            }
+
+        }
+        //time to check for a win
+        public void CheckForWin()
+        {
+            if ((MakeTowers["C"].towerBlocks.Count >= 4) || (MakeTowers["D"].towerBlocks.Count >= 4))
```

Comment:
`"D"` causes an exception to be thrown. This is the last error I investigated, so there is another one after this that causes the board to not print out all the columns correctly.

Reply:
Good catch I meant "B"....